### PR TITLE
Update to PatternFly 5.1

### PR DESCRIPTION
**Pull request by jelly**
* [x] [focus is different](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20231006-150022-64b633a0-fedora-38-storage/pixeldiff.html#TestStorageUsed-testUsed-format)
* [x] TestShutdownRestart.testBasic JavaScript error, see comments
* [x] Missing [blue border in the host switcher](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20231006-135549-64b633a0-fedora-38-expensive/pixeldiff.html#TestHostSwitching-testBasic-nav-system-rtl). This seems to be a flake, I can't reproduce it locally with our toolbox.
* [x] test/verify/check-system-services TestServices.testFilter [ND@-1] # RETRY 1 (be robust against unstable tests)`

### CODE REVIEW

**Comment by martinpitt** [comment]
node refresh collided with the xterm.js update in PR #19361, but these bitrot fast anyway, and this needs at least one fix. 

I went through all pixel diffs, and most of them are good improvements (in particular, the RTL fixes!), or some harmless spacing noise. Look at e.g. the [slider fix](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230920-203353-36740397-fedora-38-storaged/pixeldiff.html#TestStorageLuks-testLuks-format-rtl), or the [on/off switch fix](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230920-203400-36740397-fedora-38-other/pixeldiff.html#TestSelinux-testTroubleshootAlerts-selinux-rtl) both of which were the wrong way around before.

These require a more detailled look:

 - [ ] [TestStorageUsed-testUsed-format](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230920-203353-36740397-fedora-38-storaged/pixeldiff.html#TestStorageUsed-testUsed-format) gained a focus bar. Not wrong, but why didn't we have it before? Is that a timing issue?
 - [ ] [remote  host menu](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230920-162629-36740397-fedora-38-expensive/pixeldiff.html#TestHostSwitching-testBasic-nav-hosts-2-remotes-rtl) gained an extra blue bar on the selected item. Not wrong, I'm just not sure if it's intended.
 - [ ] [Reboot button on overview](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230920-203400-36740397-fedora-38-other/pixeldiff.html#TestSystemInfo-testBasic-overview-rtl) is cut off
 - [ ] [TestShutdownRestart.testBasic](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230920-203400-36740397-fedora-38-other/log.html#146-2) fails on what is probably a changed PF CSS class (`-v5` prefix perhaps?)

**Comment by garrett** [comment]
I see things that aren't "harmless". But they could be related to our RTL overrides. Here's a crop that highlights a lot of the issues:

image

Too much spacing in some form elements, no spacing for icons.

Here's another crop of the diff:

image

**Comment by garrett** [comment]
I've just pushed up a commit that removes all our custom PF RTL overrides. I wonder what the tests will look like after this. At least from what I saw, it fixed a lot of the little regressions I saw locally.

We still need some custom RTL stuff here and there (such as in graphs and on anything custom, like the metrics page), but we shouldn't need PF related overrides for RTL anymore (hopefully).

**Comment by garrett** [comment]
The popup and dropdown offset on initial draw issue has been fixed in this PR! Finally! That was so annoying.

**Comment by martinpitt** [comment]
I re-reviewed the current diffs. This completely replaces my earlier review, hiding that. Things to check:

 - [edit icon is now LTR-style](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075037-b9120bb8-fedora-38-expensive/pixeldiff.html#TestHostSwitching-testBasic-edit-hosts-rtl). [same here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestSystemInfo-testMotd-motd-rtl). [same for networking help icon](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230925-055337-b9120bb8-fedora-38-networking/pixeldiff.html#TestBonding-testBasic-networking-bond-settings-dialog-rtl).
 - [TestStorageUsed-testUsed-format](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075058-b9120bb8-fedora-38-storaged/pixeldiff.html#TestStorageUsed-testUsed-format) gained a focus bar. Not wrong, but why didn't we have it before? Is that a timing issue?
 - The [metrics expander is now the wrong way around](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestHistoryMetrics-testEvents-metrics-history-compressed-hour-rtl). We previously fixed that with an override, that one will have to come back I figure? curiously they are correct [in other places](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestSelinux-testTroubleshootAlerts-selinux-rtl), or [even got fixed by this PR](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestKeys-testPrivateKeys-ssh-keys-dialog-rtl)
 - [TestShutdownRestart.testBasic](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/log.html#146-2) fails on what is probably a changed PF CSS class (-v5 prefix perhaps?)
 - The [RTL On/Off switch on networking page](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230925-055337-b9120bb8-fedora-38-networking/pixeldiff.html#TestNetworkingBasic-testBasic-network-interface-rtl) is now the wrong way around. 

Most other changes are RTL fixes, some are harmless dialog primary button spacing changes.

File: pkg/lib/patternfly/patternfly-5-overrides.scss [hunk 1]
```diff
@@ -331,220 +331,6 @@ select.pf-v5-c-form-control {
...
-}
-
-// Override PF to fix Right to Left languages
-[dir="rtl"] {
-  // Flip outlines for buttons
-  .pf-m-split-button {
-    button::before {
-      transform: scaleX(-1);
-    }
-  }
```
Comment (jelly) [inline pkg/lib/patternfly/patternfly-5-overrides.scss:456]: The EditIcon shows that this is not fixed.

**Comment by jelly** [comment]
> I re-reviewed the current diffs. This completely replaces my earlier review, hiding that. Things to check:
> 
>     * [edit icon is now LTR-style](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075037-b9120bb8-fedora-38-expensive/pixeldiff.html#TestHostSwitching-testBasic-edit-hosts-rtl). [same here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestSystemInfo-testMotd-motd-rtl). [same for networking help icon](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230925-055337-b9120bb8-fedora-38-networking/pixeldiff.html#TestBonding-testBasic-networking-bond-settings-dialog-rtl).

Will try to reproduce with the new pre-release and report.
 
>     * [TestStorageUsed-testUsed-format](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075058-b9120bb8-fedora-38-storaged/pixeldiff.html#TestStorageUsed-testUsed-format) gained a focus bar. Not wrong, but why didn't we have it before? Is that a timing issue?

 
>     * The [metrics expander is now the wrong way around](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestHistoryMetrics-testEvents-metrics-history-compressed-hour-rtl). We previously fixed that with an override, that one will have to come back I figure? curiously they are correct [in other places](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestSelinux-testTroubleshootAlerts-selinux-rtl), or [even got fixed by this PR](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/pixeldiff.html#TestKeys-testPrivateKeys-ssh-keys-dialog-rtl)
> 
>     * [TestShutdownRestart.testBasic](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/log.html#146-2) fails on what is probably a changed PF CSS class (-v5 prefix perhaps?)

Does not seem to have changed: https://github.com/patternfly/patternfly/blob/5233e05c41cae220cf062998839f8881dfd8a080/src/patternfly/components/CalendarMonth/examples/CalendarMonth.md?plain=1#L369

>     * The [RTL On/Off switch on networking page](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230925-055337-b9120bb8-fedora-38-networking/pixeldiff.html#TestNetworkingBasic-testBasic-network-interface-rtl) is now the wrong way around.

I'll try to report that as well.

**Comment by jelly** [comment]
Reported the Switch/Icon issue on Slack. I'll wait on the tests and see if [TestShutdownRestart.testBasic](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20230921-075055-b9120bb8-fedora-38-other/log.html#146-2) fails again.

**Comment by jelly** [comment]
image

```
overview.js:11 Uncaught Error: `initialFocus` as selector refers to no known node
    at f (overview.js:11:68028)
    at u (overview.js:11:68122)
    at overview.js:11:72515
f @ overview.js:11
u @ overview.js:11
(anonymous) @ overview.js:11
```

**Comment by jelly** [comment]
> image
> 
> ```
> overview.js:11 Uncaught Error: `initialFocus` as selector refers to no known node
>     at f (overview.js:11:68028)
>     at u (overview.js:11:68122)
>     at overview.js:11:72515
> f @ overview.js:11
> u @ overview.js:11
> (anonymous) @ overview.js:11
> ```

This is likely a react-styles/patternfly-react update which went wrong. However I don't see any evidence of it.

The DatePicker is wrapped in a Popover which uses `focusTrap` and requires an element to focus on which is null in our case.
image

In the demo it is also null but the drawer shows up fine.
```
document.querySelector('.pf-v5-c-calendar-month__dates-cell.pf-m-selected'); 
```

**Comment by jelly** [comment]
For the icons PF supports:
```
<Icon shouldMirrorRTL><EditIcon /></Icon>
<EditIcon className="pf-v5-m-mirror-inline-rtl" />
```

So we either need to do this or apply our own CSS rule where fit.

**Comment by jelly** [comment]
Debugged the shutdown issue which is introduced by this [commit](https://github.com/patternfly/patternfly-react/commit/3707ffb2d8). Reverting it in node_modules no longer gives an oops.

**Review by martinpitt — APPROVED** [review]
Many thanks for pushing this through! There are [so many good RTL fixes](https://github.com/cockpit-project/pixel-test-reference/compare/a760cc815d4d51eb9364963fef10162c6d49e6ef..fcd40a3596f061917d7911a183c72143e06f14d6), code changes also LGTM.

**Comment by martinpitt** [comment]
@jelly Meh, this is still putting up a good fight..  [These](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20231010-073838-17aad30b-centos-8-stream-other/log.html#158-2) and [this](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20231010-073835-17aad30b-ubuntu-stable-other/log.html#55) are unrelated, but [this is a metrics page crash](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19360-20231010-073835-17aad30b-ubuntu-stable-other/log.html#55).

**Pull request merged**